### ISSUE TITLE
[bug] Cannot fix storage path once corrupted (as result from automation error)

### ISSUE THREAD

**Issue by Amir-Ti** [conan-io/conan#15359]
### Environment details

* Operating System+version: Windows 11 Home Version 23H2
* Compiler+version: Microsoft (R) C/C++ Optimizing Compiler Version 19.36.32535 for x64
* Conan version: 1.61.0
* Python version: 3.11.4

### Steps to reproduce

1. accidently make an error setting config `storage.path`, such as:
```python
import subprocess
path = 'C:\\'
# notice we forgot f before last string, making it a format string
subprocess.call(['conan', 'config', 'set', 'storage.path={path}\\conanStorage'])
```
2. command succeeds, but all subsequent commands fail as such:
```powershell
> conan --version
Error in Conan initialization: Conan storage path has to be an absolute path
```
even when trying to fix it:
```powershell
> conan config set storage.path=C:\conanStorage
Error in Conan initialization: Conan storage path has to be an absolute path
```
3. Manually inspecting conan.conf reveals the issue
```powershell
> Get-content $Env:CONAN_USER_HOME\.conan\conan.conf | select-string "path"

path = {path}\conanStorage
```
Expectations:
1. Either print the erroneous path (easily revealing where the issue is, perhaps with the config file path), or
2. subsequent `storage.path` commands do succeeds, thus preparing the previous mistake, or
3. fail the original config set command, so as to not "lock out" the user, forcing a manual intervention, or cache clear

**Comment by memsharded** [comment]
Hi @Amir-Ti 

Thanks for your detailed and reproducible report.

This is correct, defining an incorrect path in the ``conan.conf`` (what ``conan config set`` does), renders the Conan cache unusable (other caches are usable, defined by CONAN_USER_HOME).

The command ``conan config set`` has been removed in 2.0, and the whole ``conan.conf`` has been replaced by ``global.conf`` in 2.0 too. The storage path can be defined by editing the ``global.conf`` (which can also be shared with ``conan config install``, and it allows jinja rendering for any customization), so failures in editing the ``global.conf`` would be solved by editing back the ``global.conf``.

I think fixing this to improve the UX protecting better against this input errors sounds too much for 1.X, I will double check with the team, but I am tentatively proposing this as won't fix (outdated flow by 2.0). Thanks very much for the feedback anyway!

**Comment by Amir-Ti** [comment]
@memsharded I appreciate the **extremely** hasty response!

I didn't know about the changes in 2.x, good to know.

I wasn't thinking about changing the behavior in any way. One thing though is that the same test that checks the path upon initialization, could be run when the path is set, and simply print a warning (without it failing, as to not really affect the outcome of the command).

That way, scrolling through a log of actions would reveal a possible avenue of investigation of "when did it all go bad"?

Just a suggestion, i appreciate the responsiveness.

**Comment by memsharded** [comment]
> One thing though is that the same test that checks the path upon initialization, could be run when the path is set, and simply print a warning (without it failing, as to not really affect the outcome of the command).

Yes, that would be nice, it is just the trade-off between effort to implement that check (not that trivial), the risk of introducing possible bugs and issues, and the value provided by the warning (this is not expected to hit many users, but just a very very few), that balances to the side of not adding it, even if it could be eventually helpful for these cases. Thanks again for the feedback!

**Comment by memsharded** [comment]
Closing this ticket as outdated Conan 1.X and solved in Conan 2.0, please create new tickets for any further question or issue, thanks!
